### ISSUE TITLE
[Feature Request] JSON Input - Minify Button

### ISSUE THREAD

**Issue by barell** [getinsomnia/insomnia#1160]
- Insomnia Version: 6.0.1
- Operating System: Ubuntu 16.04

## Details

At the moment when I send a request using JSON, there is a button "Beautify JSON" at the bottom left corner. It is very useful and quick when I paste unformatted JSON to make it readable.

I think it would be nice to have opposite feature, eg. button called "Minify JSON" which basically means no new lines and spaces between JSON tokens. I guess, similar thing could be done to XML request type as I can see there is "Beautify XML" button.

**Comment by dungmv** [comment]
I created pr minify json and xml, hope that useful #4154

**Comment by dimitropoulos** [comment]
could you please provide some context as to why you need this feature?

would a plugin that does it not be sufficient in some way?

**Comment by dungmv** [comment]
a few days ago, my co-worker ask me Curl. I was generating Curl from Insomnia, it look awful. so I think, it will be better if the body minify before generating Curl, And sometimes I want to send a body to co-workers

**Comment by dimitropoulos** [comment]
Are you aware of the `Copy as Curl` feature?

### XML

![Screenshot_20211102_081643](https://user-images.githubusercontent.com/15232461/139844369-220d8846-57dd-4d7a-9cdb-043d4ad9afde.png)


```bash
curl --request POST \
  --url http://mockbin.org/echo \
  --header 'Content-Type: application/xml' \
  --data '<slideshow>
	<author>Yours Truly</author>
	<title>Sample Slide Show</title>
</slideshow>'
```

### JSON

![](https://user-images.githubusercontent.com/15232461/139843691-8c96312e-f112-4246-bc97-b1328645b493.png)

```bash
curl --request POST \
  --url http://mockbin.org/echo \
  --header 'Content-Type: application/json' \
  --data '{
  "slideshow": {
    "author": "Yours Truly",
    "title": "Sample Slide Show"
  }
}'
```

Does this solve your use-case (and more)?

**Comment by dungmv** [comment]
@dimitropoulos  I know how to generate curl from insomnia,
but you should try with a large body, it looks very bad, and when I try to send it to my co-worker by telegram, It breaks into multi-message and copies very hard
for example
```
curl --request POST \
  --url http://localhost/order_bundle/tmdt_export_order \
  --header 'Authorization: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJpYXQiOjE2MzUyMTkwNzgsImlzcyI6ImdyYXBodnRzYWxlLmN1b25nYmEuY29tIiwiYXVkIjoiZ3JhcGh2dHNhbGUuY3VvbmdiYS5jb20iLCJleHAiOjE2MzYwODMwNzgsImRhdGEiOnsiaWRlbnRpdHkiOm51bGwsImF1dGhlbnRpY2F0ZWRfdGltZSI6MTYzNTIxOTA3OCwibmFtZSI6IlZUUyBhcHBsaWNhdGlvbiIsInBhZGRpbmdfYXV0aG9yaXRpZXMiOlsiKiJdLCJ1c2VyX2lkIjo1NzcxLCJpbmZvIjpudWxsLCJhZ2VudCI6ImFwcCIsImNoYW5uZWwiOnsiaWQiOjE4MTkzLCJuYW1lIjpudWxsLCJjb21wYW55X25hbWUiOm51bGwsImNvbXBhbnlfaWQiOm51bGwsImNvbnZlcnNhdGlvbnNfZGl2aWRlIjpudWxsLCJ1c2VyY3JlYXRlIjpudWxsfSwicm9vdCI6IjhmOTI1YmEyMWI5ZTM4ODQwYjliZWQ4NjZlM2FmMDhkIn19.FPHCkC5zvz78NVOgwz8XatauPXYt-Nrh-N2yjySvuMvVs6wODW2xARJaQzxyLU0eSLEcQokUrJeoL9pBjy2HUw' \
  --header 'Content-Type: application/json' \
  --header 'Cookie: PHPSESSID=bc746535edf0aa5b741f58ee7ec08e4b' \
  --cookie PHPSESSID=bc746535edf0aa5b741f58ee7ec08e4b \
  --data '{
	"order": {
		"id": 233017
	},
	"channel_id": 18193,
	"usercreate": 5771,
	"products": [
		49528,
		49529,
		49530,
		49531,
		49532
	],
	"order_products": [
		259981,
		259982,
		259983,
		259984,
		259985,
		259986,
		259987,
		259988
	],
	"tmdt_name": "lazada"
}'
```
with minify body
```
curl --request POST \
  --url http://localhost/order_bundle/tmdt_export_order \
  --header 'Authorization: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJpYXQiOjE2MzUyMTkwNzgsImlzcyI6ImdyYXBodnRzYWxlLmN1b25nYmEuY29tIiwiYXVkIjoiZ3JhcGh2dHNhbGUuY3VvbmdiYS5jb20iLCJleHAiOjE2MzYwODMwNzgsImRhdGEiOnsiaWRlbnRpdHkiOm51bGwsImF1dGhlbnRpY2F0ZWRfdGltZSI6MTYzNTIxOTA3OCwibmFtZSI6IlZUUyBhcHBsaWNhdGlvbiIsInBhZGRpbmdfYXV0aG9yaXRpZXMiOlsiKiJdLCJ1c2VyX2lkIjo1NzcxLCJpbmZvIjpudWxsLCJhZ2VudCI6ImFwcCIsImNoYW5uZWwiOnsiaWQiOjE4MTkzLCJuYW1lIjpudWxsLCJjb21wYW55X25hbWUiOm51bGwsImNvbXBhbnlfaWQiOm51bGwsImNvbnZlcnNhdGlvbnNfZGl2aWRlIjpudWxsLCJ1c2VyY3JlYXRlIjpudWxsfSwicm9vdCI6IjhmOTI1YmEyMWI5ZTM4ODQwYjliZWQ4NjZlM2FmMDhkIn19.FPHCkC5zvz78NVOgwz8XatauPXYt-Nrh-N2yjySvuMvVs6wODW2xARJaQzxyLU0eSLEcQokUrJeoL9pBjy2HUw' \
  --header 'Content-Type: application/json' \
  --header 'Cookie: PHPSESSID=bc746535edf0aa5b741f58ee7ec08e4b' \
  --cookie PHPSESSID=bc746535edf0aa5b741f58ee7ec08e4b \
  --data '{"order":{"id":233017},"channel_id":18193,"usercreate":5771,"products":[49528,49529,49530,49531,49532],"order_products":[259981,259982,259983,259984,259985,259986,259987,259988],"tmdt_name":"lazada"}'
  ```

**Comment by develohpanda** [comment]
I see, thanks for the example. Seems reasonable but I'm not sure if it needs a minify button on the UI. Is there a case for  always generating curl with minified data, or make that somehow configurable from the UI within the "Generate Code" dialog? @wdawson 

We could also extract this to a plugin but we'll need to add some plugin APIs to make that happen.

![image](https://user-images.githubusercontent.com/4312346/140002632-c19b1870-4c9c-44d9-ad59-fe2e350f816a.png)

**Comment by wdawson** [comment]
Yeah, it seems like having the curl code generate minified by default will be a better solution for everyone rather than adding buttons and options that will make the UI more complicated. @dungmv would that work for you?

**Comment by dungmv** [comment]
@wdawson look sound good, but I prefer a button minify on UI, it is more helpful

**Comment by wdawson** [comment]
I am not sure that a button is generally helpful to enough people to justify putting one in the main app. But I think automatically minifying in the code generation is. 

If you want a button though, @dungmv, I'd suggest making a plugin.

**Comment by long76** [comment]
+1
minify `--data` block by `copy as curl`

**Comment by dimitropoulos** [comment]
just to update: we are actively working on this (but via the path of httpsnippet).  We talked about this exact ticket on the stream last week https://youtu.be/mueVITe2quw?t=1128 and this week we went over everything https://youtu.be/5Di4bmIGTrg regarding httpsnippet that's being done to unblock.

**Comment by johnwchadwick** [comment]
If minification is going to be added as a feature to httpsnippet, then I suppose it would be worth closing #4154... (and it sounds like if such functionality from that PR specifically is useful, it makes more sense to have it as a sort of plugin instead?)

**Comment by dimitropoulos** [comment]
yeah, I was planning on closing them once the httpsnippet functionality makes its way into insomnia.  I don't want to forget about it since the root issue (sending a curl command to a colleague) hasn't been fixed yet.
